packages/core/src/lib/markdown-it/plugins/markdown-it-icons.ts

```diff
@@ -1,65 +1,97 @@
+  }
+  return iconClass ? octiconIcon.toSVG({ class: iconClass }) : octiconIcon.toSVG();
+};
+
+const handleOcticonLight = (iconFontName: string, iconClass?: string) => {
+  const octiconIcon = getOcticonIcon(iconFontName);
+  if (octiconIcon === null) {
+    return '<span aria-hidden="true"></span>';
+  }
+  const octiconIconHtml = iconClass ? octiconIcon.toSVG({ class: iconClass }) : octiconIcon.toSVG();
+  const $ = cheerio.load(octiconIconHtml);
+  $('svg').attr('style', 'color: #fff');
+  return $.html();
+};
+
+const handleMaterialIcon = (iconFontType: string, iconFontName: string) => {
+  let materialIconsClass = 'material-icons';
+  switch (iconFontType) {
+  case 'mio':
+    materialIconsClass += '-outlined';
+    break;
+  case 'mir':
+    materialIconsClass += '-round';
+    break;
+  case 'mis':
+    materialIconsClass += '-sharp';
+    break;
+  case 'mit':
+    materialIconsClass += '-two-tone';
+    break;
+  default:
+  }
+  return `<span aria-hidden="true" class="${materialIconsClass} align-middle">`
+    + `${iconFontName.split('-').join('_')}</span>`;
+};
+
+const handleFontAwesome = (iconFontType: string, iconFontName: string) =>
+  `<span aria-hidden="true" class="${iconFontType} fa-${iconFontName}"></span>`;
 
-    if (iconFontType === 'glyphicon') {
-      return `<span aria-hidden="true" class="glyphicon glyphicon-${iconFontName}"></span>`;
-    } else if (iconFontType === 'octicon') {
-      const octiconIcon = getOcticonIcon(iconFontName);
-      // ensure octicons are valid
-      if (octiconIcon === null) {
-        return '<span aria-hidden="true"></span>';
-      }
-      return iconClass
-        ? octiconIcon.toSVG({ class: iconClass })
-        : octiconIcon.toSVG();
-    } else if (iconFontType === 'octiconlight') {
-      const octiconIcon = getOcticonIcon(iconFontName);
-      // ensure octicons are valid
-      if (octiconIcon === null) {
-        return '<span aria-hidden="true"></span>';
-      }
-      const octiconIconHtml = iconClass
-        ? octiconIcon.toSVG({ class: iconClass })
-        : octiconIcon.toSVG();
-      const $ = cheerio.load(octiconIconHtml);
-      $('svg').attr('style', 'color: #fff');
-      return $.html();
-    } else if (iconFontType.startsWith('mi')) {
-      let materialIconsClass = 'material-icons';
-      switch (iconFontType) {
-      case 'mio':
-        materialIconsClass += '-outlined';
-        break;
-      case 'mir':
-        materialIconsClass += '-round';
-        break;
-      case 'mis':
-        materialIconsClass += '-sharp';
-        break;
-      case 'mit':
-        materialIconsClass += '-two-tone';
-        break;
-      default:
-        // .material-icons generates 'Filled' style icons; hence, no suffix is needed for 'mif'.
-      }
-      // Use .align-middle by default to vertically-align the icon with its surrounding text (if any).
-      // Also, replace dashes (-) with underscores (_) to format the icon name properly.
```

Comment:
Are these comments no longer relevant? If not we should keep them for context

Reply:
This is still relevant, I have revert them back.